client/my-sites/checkout/checkout/web-payment-box.jsx

```diff
@@ -350,6 +352,47 @@ export class WebPaymentBox extends React.Component {
 		notices.error( errorMessage );
 	};
 
+	/**
+	 * @param {string} fieldName The name of the country select field.
+	 * @param {string} selectedCountryCode The selected country.
+	 */
+	updateSelectedCountry = ( fieldName, selectedCountryCode ) => {
+		setTaxCountryCode( selectedCountryCode );
+
+		// Apple Pay needs the country where the payment will be processed
+		// (https://developer.apple.com/documentation/apple_pay_on_the_web/applepayrequest/2951833-countrycode),
+		// so call the Paygate configuration API endpoint to get that. This
+		// information is only needed when the user clicks the button to buy
+		// something and opens the Apple Pay payment sheet, so it is somewhat
+		// wasteful to call it here (every time a new country is selected);
+		// however, it is necessary because browsers will only open a payment
+		// sheet in direct response to user input (see
+		// https://developer.mozilla.org/en-US/docs/Web/API/PaymentRequest/show),
+		// so we can't delay the opening of the payment sheet on waiting for
+		// the response to this API call.
```

Comment:
Unfortunately I wasted some time before discovering this, so I figured a code comment would be useful here for any future refactoring.